dm/master/scheduler/scheduler.go

```diff
@@ -84,7 +88,7 @@ type Scheduler struct {
 	// - recover from etcd (calling `recoverSubTasks`).
 	// delete:
 	// - remove/stop subtask by user request (calling `RemoveSubTasks`).
-	subTaskCfgs map[string]map[string]config.SubTaskConfig
+	subTaskCfgs sync.Map
```

Comment:
By the way, I would like to ask again if sync.Map is used here, is there no need to use a lock at the outside?

Reply:
other key of map is changing.

what's your meaning of "a lock at the outside"

Reply:
> Please tell me if the write operation is already locked with latch lock , why still need to change to sync.Map here ?

I figured it out, because those who don't use sync.Map will have problems with concurrent map read and write, golang will report a `fatal error: concurrent map read and map write`.

> a lock at the outside

i mean https://github.com/lance6716/dm/blob/latch/dm/master/scheduler/scheduler.go#L671

I may have missed something but i don't see what the correctness problem would be if I read and write `s.subTaskCfgs ` by calling `s.AddSubTasks` concurrently

Reply:
acquire the lock then modify the object 🤔 normal case for the usage of lock

for example calling s.AddSubTasks concurrently with same name